ir/src/constraint_builder/boundary_constraints.rs

```diff
@@ -72,7 +72,7 @@ impl ConstraintBuilder {
                );
 
                 // add its expression to the constraints graph.
-                let rhs = self.insert_expr(constraint.value())?;
+                let rhs = self.insert_expr(constraint.value().clone())?;
```

Comment:
Since we're changing to a consume pattern, this clone shouldn't be necessary. Instead, at the top of this function we should have the following:

```
let (boundary, access, value) = constraint.into_parts();
```

Then this will just be:
```
let rhs = self.insert_expr(value)?;
```